imageGenSD.py

```diff
@@ -532,88 +563,146 @@ async def roll(interaction: dis.Interaction):
 
-        await interaction.response.send_message('Select a profile to view:',view=view)
+            await interaction.response.send_message(f'Please close your existing dropdown menu or wait for it to time out.', ephemeral=True, delete_after=9.0)
+
+        else:
+
+            dis_log.debug(f"Creating a SHOW PROFILE view for user {interaction.user.id}.")
+            user_id  = interaction.user.id if user == None else user.id
+            profiles = db_ifc.getProfiles(user_id)
+            dis_log.debug(f"User {user_id} returned profile list {profiles}.")
+            view = ddf.DropdownView(ctx      = interaction,
+                                    type     = ddf.DropDownTypeEnum.SHOW,
+                                    choices  = profiles,
+                                    metadata = metadata)
+
+            await interaction.response.send_message('Select a profile to view:',view=view)
 
     else:
 
         opts = {'id' : profile_id}
 
         dis_log.debug(f"Creating a job with metadata {metadata} and options {opts}.")
-        job = jf.JobFactory.getJob(type    = jf.JobTypeEnum.SHOWPROFILE,
+        job = jf.JobFactory.getJob(type    = jf.JobTypeEnum.SHOW_PROFILE,
                                    ctx     = interaction,
                                    options = opts)
         dis_log.debug(f"Posting SHOW job {job} to the queue.")
-        result = job_queue.add(metadata = metadata,
-                               job      = job)
+        result = show_queue.add(metadata = metadata,
+                                job      = job)
 
         await interaction.response.send_message(f'{result}', ephemeral=True, delete_after=9.0)
 
-#This is commented until the failed inheritance issue can be resolved.
-#@IGSD_client.tree.command()
-#async def testclear(interaction: dis.Interaction):
-#    """A test flood of requests to verify the flush command works.
-#
-#       Input  : None.
-#
-#       Output : None.
-#    """
-#    dis_log = log.getLogger('discord')
-#    rng    = range(1,10)
-#
-#    job = [{ 'metadata' : {
-#                'ctx'    : interaction,
-#                'loop'   : IGSD_client.getLoop(),
-#                'poster' : post
-#           },
-#           'data' : {
-#                #Requests are sorted by guild for rate-limiting
-#                'guild'  : interaction.guild_id,
-#                #This should really be metadata but the rest of the metadata
-#                #can't be pickeled, so this must be passed with the work.
-#                'id'     : x,
-#                'post'   : {'empty'},
-#                'reply'  : "test job"
-#            }
-#        } for x in rng]
-#    dis_log.debug(f"trying to clear the queue.")
-#
-#    for m in rng:
-#        result = job_queue.add(job[m -1])
-#
-#    job_queue.flush()
-#
-#    await interaction.response.send_message(f'{result}', ephemeral=True, delete_after=9.0)
+
+@verify(UNIQUE)
+class SummaryChoices(Enum):
```

Comment:
Why not make it an IntEnum?

Reply:
Mostly because it didn't need to be; arguably the other `IntEnums` don't really need to be either (since `IntEnum` is the specialized class and `Enum` is the general).